### ISSUE TITLE
Discord fails to evaluate JS due to CSP

### ISSUE THREAD

**Issue by jdm** [servo/servo#34635]
After successfully logging in, this error appears in the console and an error message appears in the client:
```
(new EvalError("call to Function() blocked by CSP", "https://discord.com/assets/web.3293dbbf90bafac60d3c.js", 12))
```
This corresponds to [this hook](https://github.com/servo/servo/blob/a37a9f68b810e0e7d2a2d4cce4b9e5706288c434/components/script/script_runtime.rs#L320) returning false.

**Comment by jdm** [comment]
The CSP list:
```
csp list: Some(CspList([Policy { directive_set: [Directive { name: "default-src", value: ["'self'"] }, Directive { name: "script-src", value: ["'self'", "'unsafe-eval'", "'unsafe-inline'", "'nonce-NjAsMjA2LDE5NSwxMjUsMTIxLDI0NCw2LDY2'", "blob:", "https://cdn.discordapp.com/animations/", "https://www.gstatic.com/recaptcha/", "https://www.google.com/recaptcha/", "https://recaptcha.net/recaptcha/", "https://*.hcaptcha.com", "https://hcaptcha.com", "https://js.stripe.com", "https://js.braintreegateway.com", "https://assets.braintreegateway.com", "https://www.paypalobjects.com", "https://checkout.paypal.com", "https://c.paypal.com", "https://kit.cash.app", "https://static.discord.com", "https://static-edge.discord.com"] }, Directive { name: "style-src", value: ["'self'", "'unsafe-inline'", "https://cdn.discordapp.com", "https://*.hcaptcha.com", "https://hcaptcha.com", "https://kit.cash.app", "https://static.discord.com", "https://static-edge.discord.com"] }, Directive { name: "img-src", value: ["'self'", "blob:", "data:", "https://*.discordapp.net", "https://*.discordapp.com", "https://*.discord.com", "https://i.scdn.co", "https://i.ytimg.com", "https://i.imgur.com", "https://media.tenor.co", "https://media.tenor.com", "https://c.tenor.com", "https://*.youtube.com", "https://*.giphy.com", "https://static-cdn.jtvnw.net", "https://pbs.twimg.com", "https://assets.braintreegateway.com", "https://checkout.paypal.com", "https://c.paypal.com", "https://*.stats.paypal.com", "https://api.cash.app"] }, Directive { name: "font-src", value: ["'self'", "https://fonts.gstatic.com", "https://cash-f.squarecdn.com", "https://static.discord.com", "https://static-edge.discord.com"] }, Directive { name: "connect-src", value: ["'self'", "https://status.discordapp.com", "https://status.discord.com", "https://support.discordapp.com", "https://support.discord.com", "https://discordapp.com", "https://discord.com", "https://discord-attachments-uploads-prd.storage.googleapis.com", "https://cdn.discordapp.com", "https://media.discordapp.net", "https://images-ext-1.discordapp.net", "https://images-ext-2.discordapp.net", "https://router.discordapp.net", "wss://*.discord.gg", "https://best.discord.media", "https://latency.discord.media", "wss://*.discord.media", "wss://dealer.spotify.com", "https://api.spotify.com", "https://music.amazon.com/embed/oembed", "https://*.sentry.io", "https://api.twitch.tv", "https://api.stripe.com", "https://api.braintreegateway.com", "https://client-analytics.braintreegateway.com", "https://*.braintree-api.com", "https://www.googleapis.com", "https://*.algolianet.com", "https://*.hcaptcha.com", "https://hcaptcha.com", "https://*.algolia.net", "ws://127.0.0.1:*", "http://127.0.0.1:*"] }, Directive { name: "media-src", value: ["'self'", "blob:", "disclip:", "https://*.discordapp.net", "https://*.discord.com", "https://*.discordapp.com", "https://*.youtube.com", "https://streamable.com", "https://vid.me", "https://twitter.com", "https://oddshot.akamaized.net", "https://*.giphy.com", "https://i.imgur.com", "https://media.tenor.co", "https://media.tenor.com", "https://c.tenor.com"] }, Directive { name: "frame-src", value: ["https://discordapp.com/domain-migration", "discord:", "https://www.google.com/recaptcha/", "https://recaptcha.net/recaptcha/", "https://*.hcaptcha.com", "https://hcaptcha.com", "https://js.stripe.com", "https://hooks.stripe.com", "https://checkout.paypal.com", "https://c.paypal.com", "https://assets.braintreegateway.com", "https://checkoutshopper-live.adyen.com", "https://kit.cash.app", "https://player.twitch.tv", "https://clips.twitch.tv/embed", "https://player.vimeo.com", "https://www.youtube.com/embed/", "https://www.tiktok.com/player/", "https://music.amazon.com/embed/", "https://music.amazon.co.uk/embed/", "https://music.amazon.de/embed/", "https://music.amazon.co.jp/embed/", "https://music.amazon.es/embed/", "https://music.amazon.fr/embed/", "https://music.amazon.it/embed/", "https://music.amazon.com.au/embed/", "https://music.amazon.in/embed/", "https://music.amazon.ca/embed/", "https://music.amazon.com.mx/embed/", "https://music.amazon.com.br/embed/", "https://www.youtube.com/s/player/", "https://twitter.com/i/videos/", "https://www.funimation.com/player/", "https://www.redditmedia.com/mediaembed/", "https://open.spotify.com/embed/", "https://w.soundcloud.com/player/", "https://audius.co/embed/", "https://*.watchanimeattheoffice.com", "https://sessionshare.sp-int.playstation.com/", "https://session-share.playstation.com/", "https://localhost:*", "https://*.discordsays.com", "https://discordappcom.cloudflareaccess.com/", "https://d3ogqhtsivkon3.cloudfront.net/"] }, Directive { name: "child-src", value: ["'self'", "blob:", "https://assets.braintreegateway.com", "https://checkout.paypal.com", "https://c.paypal.com"] }, Directive { name: "prefetch-src", value: ["'self'", "https://cdn.discordapp.com/assets/"] }], disposition: Enforce, source: Header }]))
```

It doesn't look like it should prevent any script execution, so I suspect our logic is buggy.

**Comment by jdm** [comment]
Filed https://github.com/rust-ammonia/rust-content-security-policy/issues/48.

**Comment by jdm** [comment]
Fixed by #34723.
